vsock-proxy/parent/src/parent.rs

```diff
@@ -751,4 +774,54 @@ mod tests {
             check_create_rw_block_file_res(testcase.0, testcase.2, testcase.3);
         }
     }
+
+    #[test]
+    fn test_runtime_hostname_is_not_removed() {
+        let mut runtime_env_vars = HashMap::new();
+        runtime_env_vars.insert("HOSTNAME".to_string(), "my-new-host".to_string());
+
+        filter_parent_env_from_runtime_envs(&mut runtime_env_vars, ("HOSTNAME", "my-old-host"));
+
+        // Ensure the entry remains unchanged
+        assert_eq!(
+            runtime_env_vars.get("HOSTNAME"),
+            Some(&"my-new-host".to_string())
+        );
+    }
+
+    #[test]
+    fn test_runtime_path_key_is_removed() {
+        let mut runtime_env_vars = HashMap::new();
+        runtime_env_vars.insert("PATH".to_string(), "old_value".to_string());
+
+        filter_parent_env_from_runtime_envs(&mut runtime_env_vars, ("PATH", "new_value"));
+
+        // Ensure the key was removed
+        assert!(runtime_env_vars.get("PATH").is_none());
+    }
+
+    #[test]
+    fn test_runtime_key_not_removed_when_values_dont_match() {
+        let mut runtime_env_vars = HashMap::new();
+        runtime_env_vars.insert("MY_VAR".to_string(), "new_value".to_string());
+
+        filter_parent_env_from_runtime_envs(&mut runtime_env_vars, ("MY_VAR", "old_value"));
+
+        // Ensure the key was not removed
+        assert_eq!(
+            runtime_env_vars.get("MY_VAR"),
+            Some(&"new_value".to_string())
+        );
+    }
+
+    #[test]
+    fn test_runtime_key_not_removed_when_not_present() {
+        let mut runtime_env_vars = HashMap::new();
+        runtime_env_vars.insert("RUNTIME_VAR".to_string(), "runtime_value".to_string());
```

Comment:
MY_VAR should also be inserted with "value" as the value. We should check that it is removed on line 825.